### ISSUE TITLE
Feature: SLM Training Pipeline – Channel Q&A Import → Fine-Tune → Human Eval UI → Feedback Loop

### ISSUE THREAD

**Issue by voipin** [cloudwarriors-ai/cw-openclaw#9]
## Overview

Build an end-to-end training pipeline that ingests **Q&A data from Zoom channel imports**, uses it to fine-tune a **Small Language Model (SLM)**, surfaces the model's outputs through a **human evaluation UI**, and feeds reviewer feedback back into the training system for continuous improvement.

### Stage 1: Data Ingestion (Channel Q&A Import)
- [ ] Import Q&A pairs from Zoom Team Chat channel history (existing channel import tooling)
- [ ] Parse and normalize raw channel messages into structured Q&A format:
  ```json
  {
    "question": "How do I reset a Zoom Phone user?",
    "answer": "Go to Admin > Phone System > Users, select the user, and click Reset.",
    "source_channel": "zoom-support",
    "timestamp": "2026-02-23T15:00:00Z",
    "confidence": null
  }
  ```
- [ ] Filter low-quality pairs (too short, off-topic, incomplete answers)
- [ ] Deduplication of near-identical Q&A pairs
- [ ] Store in a training dataset store (Postgres or flat JSONL files)

### Stage 2: SLM Fine-Tuning
- [ ] Select base SLM (e.g., Phi-3-mini, Mistral 7B, or similar small model)
- [ ] Format dataset into fine-tuning format (instruction-tuning / chat format)
- [ ] Training job runner — trigger fine-tune run on new data batches
- [ ] Version training runs — track dataset version, hyperparams, and model checkpoint per run
- [ ] Evaluate baseline metrics post-training (BLEU, ROUGE, or custom eval)

### Stage 3: Human Evaluation UI
- [ ] Web UI for reviewers to evaluate model outputs
- [ ] For each Q&A pair, show:
  - The original question
  - The SLM's generated answer
  - The reference answer (from channel import)
- [ ] Reviewer actions:
  - ✅ Approve (answer is correct/good)
  - ✏️ Edit (correct the answer inline)
  - ❌ Reject (answer is wrong/harmful)
  - ⭐ Rate quality (1–5 scale)
  - 💬 Add a comment/note
- [ ] Reviewer assignment — route specific Q&A batches to specific reviewers
- [ ] Progress tracking dashboard — how many reviewed, pending, approved, rejected
- [ ] Auth — restrict to internal team members

### Stage 4: Feedback Loop (RLHF-lite)
- [ ] Collect reviewer edits and ratings as labeled training signal
- [ ] Merge approved + edited answers back into the training dataset
- [ ] Rejected pairs flagged and excluded from future training runs
- [ ] Trigger re-training job when feedback batch reaches threshold (e.g., 100 new labels)
- [ ] Track model improvement over iterations (eval score history)
- [ ] Diff view: compare model performance before/after each feedback round

## Data Flow Diagram

```
Channel Messages
      │
      ▼
Q&A Extraction & Normalization
      │
      ▼
Training Dataset (versioned)
      │
      ▼
SLM Fine-Tune Job
      │
      ▼
Model Checkpoint
      │
      ▼
Human Eval UI ──► Reviewer Feedback
      │                    │
      └────────────────────┘
              │
              ▼
      Updated Training Dataset
              │
              ▼
         Next Training Run
```

## Tech Stack (Suggested)

| Component | Options |
|-----------|---------|
| Base SLM | Phi-3-mini, Mistral 7B, Llama 3.2 |
| Fine-tuning | Hugging Face Transformers + PEFT/LoRA |
| Dataset store | JSONL files + Postgres metadata |
| Eval UI | Next.js or simple React app |
| Training runner | Python script / Modal / RunPod |
| Feedback DB | Postgres |

## Requirements

- [ ] Channel import adapter (connects to existing moltbot channel Q&A ingestion)
- [ ] Dataset versioning and lineage tracking
- [ ] Fine-tuning pipeline (scriptable, repeatable)
- [ ] Human eval UI (web app)
- [ ] Feedback ingestion API
- [ ] Re-training trigger (threshold-based or manual)
- [ ] Model eval scoring pre/post each round
- [ ] Admin dashboard: dataset stats, training run history, eval scores

## Integration Points

- **Issue #4 (Stakeholders)** — notify stakeholders when a training run completes or eval batch is ready for review
- **Issue #3 (DM Reminders)** — remind reviewers of pending eval tasks via Zoom DM
- Existing channel import / memory system in moltbot-workspace

## Notes

- Start with a small curated dataset from the highest-quality channels
- LoRA/QLoRA fine-tuning recommended to keep compute costs low
- Human eval UI should be dead simple — reviewers shouldn't need training to use it
- Consider using the eval UI feedback to also improve the **channel prefilter** classifier (meta-training)
- Long-term: automated eval using a judge LLM to reduce human reviewer burden

---
*Replicated from [cloudwarriors-ai/moltbot-workspace#5](https://github.com/cloudwarriors-ai/moltbot-workspace/issues/5)*
